src/ServiceControl.UnitTests/ApprovalFiles/APIApprovals.ApproveDomainEventTypes.approved.txt

```diff
@@ -0,0 +1,288 @@
+namespace ServiceControl.Contracts.CustomChecks
+{
+    public class CustomCheckFailed : ServiceControl.Infrastructure.DomainEvents.IDomainEvent, ServiceControl.Infrastructure.SignalR.IUserInterfaceEvent
+    {
+        public CustomCheckFailed() { }
+        public string Category { get; set; }
+        public string CustomCheckId { get; set; }
+        public System.DateTime FailedAt { get; set; }
+        public string FailureReason { get; set; }
+        public System.Guid Id { get; set; }
+        public ServiceControl.Contracts.Operations.EndpointDetails OriginatingEndpoint { get; set; }
```

Comment:
Hmmm. We need to find a way to find all of the child types too as they form a part of the contract. I wonder if the ApiApprovals library is the best way to do this or if some basic reflection code would be better suited to our needs here.

Reply:
For instance, we probably don't care about interfaces or constructors for this approval.

Reply:
To clarify, the content of this file is created by PublicApiGenerator, not Particular.Approvals. If this file has unwanted stuff in it, the call to `ApiGenerator.GeneratePublicApi` is what needs to be tweaked/replaced, not `Approver.Verify`.